docs/notebooks/model-levels.ipynb

```diff
@@ -0,0 +1,2052 @@
+    "\n",
+    "The [metview package](https://metview.readthedocs.io/en/latest/index.html) can be difficult to install. [Installation with conda](https://metview.readthedocs.io/en/latest/install.html#conda) may be the easiest way to get started. This notebook assumes a working installation of metview."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "\n",
+    "import cdsapi\n",
+    "import matplotlib.pyplot as plt\n",
+    "import metview as mv\n",
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "import xarray as xr\n",
+    "\n",
+    "from pycontrails import Fleet, Flight, MetDataset\n",
+    "from pycontrails.datalib.ecmwf import PRESSURE_LEVEL_VARIABLES, SURFACE_VARIABLES\n",
+    "from pycontrails.models.cocip import Cocip\n",
+    "from pycontrails.models.humidity_scaling import HistogramMatching\n",
+    "from pycontrails.models.ps_model import PSFlight"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Define CDS model elvel request\n",
+    "\n",
+    "We make three requests to the CDS API.\n",
+    "\n",
+    "- `ml_request` is a request for the model levels meteorology data\n",
+    "- `lnsp_request` is a request for the surface pressure data needed for interpolating from model levels to pressure levels\n",
+    "- `sl_request` is a request for the single level top of atmosphere radiation data needed for CoCiP"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "date = \"2024-01-15\"\n",
+    "time = list(range(18))\n",
+    "grid = \"1/1\"\n",
+    "\n",
+    "ml_variables = [\"t\", \"q\", \"u\", \"v\", \"w\", \"ciwc\"]\n",
+    "sl_variables = [\"tsr\", \"ttr\"]\n",
+    "\n",
+    "ml_request = {\n",
+    "    \"levtype\": \"ml\",\n",
+    "    \"levellist\": \"70/to/90/by/1\",\n",
+    "    \"param\": ml_variables,\n",
+    "    \"date\": date,\n",
+    "    \"time\": time,\n",
+    "    \"grid\": grid,\n",
+    "}\n",
+    "\n",
+    "lnsp_request = {\n",
+    "    \"levtype\": \"ml\",\n",
+    "    \"levelist\": \"1\",\n",
+    "    \"param\": \"lnsp\",\n",
+    "    \"date\": date,\n",
+    "    \"time\": time,\n",
+    "    \"grid\": grid,\n",
+    "}\n",
+    "\n",
+    "sl_request = {\n",
+    "    \"levtype\": \"sfc\",\n",
+    "    \"product_type\": \"reanalysis\",\n",
+    "    \"param\": sl_variables,\n",
+    "    \"date\": date,\n",
+    "    \"time\": time,\n",
+    "    \"grid\": grid,\n",
+    "    \"format\": \"netcdf\",\n",
+    "}"
+   ]
+  },
```

Comment:
Maybe I'm wrong, but I think the CDS API can deliver model-level data in netcdf format. Is there a reason not to ask for model- and single-level data in the same format?

Reply:
Yeah -- this is a great question. I think we _could_ request data in the same format, but it was a little easier to get model level data as grib and radiation data as netcdf. Essentially, `metview` has more functionality for grib -- you can read a grib file directly into a `metview.Fieldset` and this is data type needed for the ML to PL conversion. I had other issues requesting the surface level data as grib, so I opted for netcdf there. I think it would be possible to download only grib or only netcdf ... but it would require spending a little more time figuring out `metview`.

Aside: Google also offers [ARCO ERA5 data](https://cloud.google.com/storage/docs/public-datasets/era5), some of which is computed on model levels. I think a notebook showing how to integrate ARCO ERA5 into pycontrails would be way more instructive than this notebook! The ARCO ERA5 data is already "zarrified" and sitting on GCS, so it's easier to ingest compared to requesting data from CDS.